kirin/core/merge_utils.py

```diff
@@ -360,37 +360,74 @@ def log_stu_modif(trip_update, stu, string_additional_info):
     )
 
 
+def manage_arrival_consistency(stu, trip_update, previous_stop_event):
+    if stu.arrival is not None:
+        return True
+
+    logger = logging.getLogger(__name__)
+
+    stu.arrival = stu.departure
+    if stu.arrival is None and previous_stop_event.time is not None:
+        stu.arrival = previous_stop_event.time
+    if stu.arrival is None:
+        logger.warning(
+            "TripUpdate on navitia vj {nav_id} on {date} rejected: "
+            "StopTimeUpdate missing arrival time".format(
+                nav_id=trip_update.vj.navitia_trip_id, date=trip_update.vj.get_circulation_date()
+            )
+        )
+        return False
+    log_stu_modif(trip_update, stu, "arrival = {v}".format(v=stu.arrival))
+    if not stu.arrival_delay and stu.departure_delay:
+        stu.arrival_delay = stu.departure_delay
+        log_stu_modif(trip_update, stu, "arrival_delay = {v}".format(v=stu.arrival_delay))
+
+    return True
+
+
+def manage_departure_consistency(stu, trip_update):
+    if stu.departure is not None:
+        return
+    stu.departure = stu.arrival
+    log_stu_modif(trip_update, stu, "departure = {v}".format(v=stu.departure))
+    if not stu.departure_delay and stu.arrival_delay:
+        stu.departure_delay = stu.arrival_delay
+        log_stu_modif(trip_update, stu, "departure_delay = {v}".format(v=stu.departure_delay))
+
+
+def manage_deleted_stop_event(event_type, stu, previous_stop_event, trip_update):
+    # not considering deleted {event_type} (event_type can be departure or arrival)
+    if stu.is_stop_event_deleted(event_type):
+        return previous_stop_event
+    delay_field = "{}_delay"
```

Comment:
Do it only once (and remove other `format()`:
```suggestion
    delay_field = "{}_delay".format(event_type)
```